### DISCUSSION TITLE
MCP Config - make it parse and replace VSCode Variables like

### DISCUSSION THREAD

**Discussion by dkmaker** [cline/cline#925] [Feature Requests]
Would love to see the Cline MCP Config replace VSCode variables like ${workspaceFolder} with the workspace folder

A solution could be
- Resolve the first folder in the workspace opened
- If non opend dont start the server
- Reference different folders by ${workspaceFolder:folder2}/sub folder

In this way you can build MCP Servers that follows the Repo opened and will be very usefill for saving state etc. across multiple projects

**Comment by dkmaker** [comment]
![image](https://github.com/user-attachments/assets/851c7d1c-b5c0-4db1-85b6-79f75f101dd1)

I have implemented it in this repository [zenturacp/cline](https://github.com/zenturacp/cline/tree/mcp_vscode_vars) already, and tested it and it works here is an example

```json
    "git": {
      "command": "uvx",
      "args": [
        "mcp-server-git",
        "--repository",
        "${workspaceFolder}"
      ]
    }
```

**Reply by dpkirchner** [reply]
Is this something that needs to be updated for each MCP or is the variable substitution supposed to be handled globally? Here's my mcp settings, in my attempt to use the filesystem MCP:

```
{
  "mcpServers": {
    "github.com/modelcontextprotocol/servers/tree/main/src/filesystem": {
      "autoApprove": [
        "read_multiple_files",
        "list_directory",
        "directory_tree",
        "search_files",
        "get_file_info",
        "list_allowed_directories"
      ],
      "disabled": false,
      "timeout": 60,
      "command": "npx",
      "args": [
        "-y",
        "@modelcontextprotocol/server-filesystem",
        "${workspaceFolder}"
      ],
      "transportType": "stdio"
    }
  }
}
```

I get the error: `Error accessing directory ${workspaceFolder}: Error: ENOENT: no such file or directory, stat '${workspaceFolder}' at async`

FWIW the link to the zenturacp/cline repository throws a 404.

**Comment by srigi** [comment]
I just resolved the programmatic unavailability of `workspaceFolder` value in MCP code with the most stupid solution - I added a new argument to my MCP tool:
- `workspacePath` The current workspace absolute path

The LLM (Cline) fills the value for me automatically
